src/3_branch.js

```diff
@@ -301,8 +310,28 @@ Branch.prototype['init'] = wrap(callback_params.CALLBACK_ERR_DATA, function(done
 		done(err, data && utils.whiteListSessionData(data));
 	};
 
+	var allTheWaysToSayHidden =
+		[{ 'hidden': 'visibilitychange' },
+		{ 'mozHidden': 'mozvisibilitychange' },
+		{ 'msHidden': 'msvisibilitychange' },
+		{ 'webkitHidden': 'webkitvisibilitychange' }],
+
+	attachVisibilityEvent = function(data) {
+		for(var index = 0; index < allTheWaysToSayHidden.length; index++) {
+				var hidden = Object.keys(allTheWaysToSayHidden[index])[0],
+				event = allTheWaysToSayHidden[index][hidden];
+			if (typeof document[hidden] !== 'undefined') {
+				document['addEventListener'](event, function() {
+					if (!document[hidden]) { check_has_app(data, finishInit); }
```

Comment:
this isn't going to work, hidden is in the wrong closure so it won't get captured properly... need an anoymous funciton :-( (alternately you could just spell out each visibility event separately)